### PR TITLE
FISH-1021 Add support for HSTS header

### PR DESCRIPTION
**Pull request by Cousjava**
## Description
This is a feature

### New tests
New test included in PR

### Testing Performed
Go to admin console and enable HSTS from SSL page of http-listner-2, `asadmin restart-http-listeners` then curl localhost:8181 with curl and check the headers.

### Testing Environment
Zulu JDK 1.8_272 on Ubuntu 20.10 with Maven 3.6.3

### CODE REVIEW

**Comment by Cousjava** [comment]
Jenkins test please

**Comment by Cousjava** [comment]
Jenkins test please

**Comment by Cousjava** [comment]
Jenkins test please

**Comment by Cousjava** [comment]
Jenkins test please

File: nucleus/grizzly/config/src/main/java/org/glassfish/grizzly/config/GenericGrizzlyListener.java [hunk 1]
```diff
@@ -505,15 +505,14 @@ protected void configureSubProtocol(final ServiceLocator habitat,
...
-                        
                         subProtocolFilterChainBuilder.add(extraSslPUFilter);
                         final FilterChainBuilder extraSslPUFilterChainBuilder =
                             extraSslPUFilter.getPUFilterChainBuilder();
                         
                         try {
                             // temporary add SSL Filter, so subprotocol
                             // will see it
-                            extraSslPUFilterChainBuilder.add(addedSSLFilter);
+                            extraSslPUFilterChainBuilder.add(addedSSLFilter);;
```
Comment (dmatej) [inline nucleus/grizzly/config/src/main/java/org/glassfish/grizzly/config/GenericGrizzlyListener.java:515]: just single semicolon, please, your checkstyle plugin ;-)

**Comment by Cousjava** [comment]
Jenkins test please

File: nucleus/grizzly/config/src/main/java/org/glassfish/grizzly/config/GenericGrizzlyListener.java [hunk 2]
```diff
@@ -787,6 +787,8 @@ protected void configureHttpProtocol(final ServiceLocator habitat,
         configureWebSocketSupport(habitat, networkListener, http, filterChainBuilder);
 
         configureAjpSupport(habitat, networkListener, http, filterChainBuilder);
+
```
Comment (lprimak) [inline nucleus/grizzly/config/src/main/java/org/glassfish/grizzly/config/GenericGrizzlyListener.java:790]: extra blank space

File: nucleus/grizzly/config/src/test/java/org/glassfish/grizzly/config/HSTSFilterTest.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        FilterChainContext context = new FilterChainContext();
+        context.setMessage(HttpContent.create(new TestHttpHeader()));
+        
+        filter.handleRead(context);
+        return ((HttpContent) context.getMessage()).getHttpHeader();
+        
+    }
+    
+    private class TestHttpHeader extends HttpHeader {
+
```
Comment (lprimak) [inline nucleus/grizzly/config/src/test/java/org/glassfish/grizzly/config/HSTSFilterTest.java:100]: extra blank space

**Review by lprimak — APPROVED** [review]
Just minor spacing nits

**Pull request merged**